### ISSUE TITLE
Using Amazon DocumentDB as an aws_dms_endpoint

### ISSUE THREAD

**Issue by rmorris1218** [terraform-providers/terraform-provider-aws#7194]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Allow for using Amazon DocumentDB as a AWS DMS Target Endpoint as outlined [here](https://docs.aws.amazon.com/dms/latest/userguide/target.docdb.html).

### New or Affected Resource(s)

* aws_dms_endpoint

### Potential Terraform Configuration

```hcl
resource "aws_dms_endpoint" "docdb" {
  endpoint_id                 = "test-docdb-endpoint-tf"
  endpoint_type               = "target"
  engine_name                 = "docdb"
  port                        = 27017
  server_name                 = "test"
  ssl_mode                    = "verify-full"
  username                    = "test"
  password                    = "test"
}
```

### References

* https://docs.aws.amazon.com/dms/latest/userguide/target.docdb.html
* https://docs.aws.amazon.com/dms/latest/userguide/target.docdb.tutorial.html

**Comment by bflad** [comment]
Support for the `docdb` engine name has been merged and will release with version 1.59.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
